src/hera/workflows/user_container.py

```diff
@@ -37,10 +37,8 @@ def _build_image_pull_policy(self) -> Optional[str]:
             # the following 2 are "normal" entries
             **{ipp.name: ipp for ipp in ImagePullPolicy},
             **{ipp.value: ipp for ipp in ImagePullPolicy},
-            # some users might submit the policy without underscores
-            **{ipp.value.lower().replace("_", ""): ipp for ipp in ImagePullPolicy},
-            # some users might submit the policy in lowercase
-            **{ipp.name.lower(): ipp for ipp in ImagePullPolicy},
```

Comment:
These were mostly redundant and created duplicate key-value pairs.

Reference:
```python3 
# src/hera/workflows/models/io/k8s/api/core/v1.py

class ImagePullPolicy(Enum):
    always = "Always"
    never = "Never"
    if_not_present = "IfNotPresent"
```

Reply:
Good find, thanks! cc @flaviuvadan JFYI 👀